packages/eas-cli/src/commands/fingerprint/compare.ts

```diff
@@ -170,6 +174,186 @@ export default class FingerprintCompare extends EasCommand {
   }
 }
 
+function capitalizeFirstLetter(string: string): string {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+async function getFirstFingerprintInfoAsync(
+  graphqlClient: ExpoGraphqlClient,
+  projectId: string,
+  { buildId: buildIdFromArg, hash1, nonInteractive }: FingerprintCompareFlags
+): Promise<{ fingerprint: Fingerprint; platforms: AppPlatform[]; origin: string }> {
+  if (hash1) {
+    const fingerprintFragment = await getFingerprintFragmentFromHashAsync(
+      graphqlClient,
+      projectId,
+      hash1
+    );
+    const fingerprint = await getFingerprintAsync(fingerprintFragment);
+    return {
+      fingerprint,
+      platforms: inferPlatformsFromSource(fingerprint),
+      origin: `hash ${hash1}`,
+    };
+  }
+
+  let buildId: string | null = buildIdFromArg ?? null;
+  if (!buildId) {
+    if (nonInteractive) {
+      throw new Error('Build ID must be provided in non-interactive mode');
+    }
+
+    const displayName = await getDisplayNameForProjectIdAsync(graphqlClient, projectId);
+    buildId = await selectBuildToCompareAsync(graphqlClient, projectId, displayName, {
+      filters: { hasFingerprint: true },
+    });
+    if (!buildId) {
+      throw new Error(); // exit, explanation already printed in prompt
```

Comment:
```suggestion
      throw new Error('Must select build with fingerprint for comparison.');
```
so the output will have both:
```
No fingerprints have been computed for builds of project ${projectDisplayName}.
Must select build with fingerprint for comparison.
```

(this matches the exit pattern for similar cases in other commands I believe)